homeassistant/components/watergate/coordinator.py

```diff
@@ -27,9 +49,31 @@ def __init__(self, hass: HomeAssistant, api: WatergateLocalApiClient) -> None:
         )
         self.api = api
 
-    async def _async_update_data(self) -> DeviceState:
+    async def _async_update_data(self) -> WatergateAgregatedRequests:
+        # Question: Currently we are assuming that to make sure that whole device works we should have all data fetched.
+        # Is it a good approach? Should we fetch data separately?
+        # When we will change it to fetch data separately, we should also move that entity unavailaibility mechanism to entities, right?
```

Comment:
Depends on how you want it to work, do you want to fetch different data at different itnervals? Can a call fail while the rest works? Maybe it makes sense if you have separate coordinators for each, but that depends on what info the devices need

Reply:
Exectly those were my concers, and I think that I want to check them all at once and If one will fail that means that the device doesn't work, so probably I will leave it as it is. I just needed that confirmation that it is fine from experts point of view ;)